### PR TITLE
[PGA] consolidate multitool and borges-indexer into pga-create

### PR DESCRIPTION
**Pull request by smola**
* Create a new `pga-create` tool that consolidates `multitool`, `borges-indexer` and `set-forks`. This is the single entry point (other than borges) to generate PGA.
* Removed `get-index` and `get-dataset` subcommands, since we already have the `pga` cli for that purpose.

This is the first of a series of maintenance PRs to simplify the process of generating the PGA dataset.

### CODE REVIEW

**Comment by smola** [comment]
Given the monstrous amount of files changed in this PR because of vendor directory change, I suggest going to the commit list and review independent commits, except the vendor directory update one.

File: PublicGitArchive/pga-create/cmd/pga-create/main.go [hunk 1]
```diff
@@ -51,3 +52,8 @@ func main() {
 		panic(err)
 	}
 }
+
+func fail(operation string, err error) {
```
Comment (campoy) [inline PublicGitArchive/pga-create/cmd/pga-create/main.go]: I would expect this function to be used somewhere? Maybe instead of `panic(err)`?

File: PublicGitArchive/pga-create/cmd/pga-create/select.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	MinStars        int    `short:"m" long:"min-stars" description:"Minimum number of stars."`
+	Max             int    `short:"n" long:"max" default:"-1" description:"Maximum number of top-starred repositories to clone. -1 means unlimited. Language filter is applied before."`
+	FilterLanguages string `short:"l" long:"filter-languages" description:"Comma separated list of languages."`
+	UrlTemplate     string `long:"url-template" default:"git://github.com/%s.git" description:"Output URL printf template."`
+}
+
+func (c *selectCommand) Execute(args []string) error {
+	var filteredLangsSplitted []string
+	if len(c.FilterLanguages) == 0 {
+		filteredLangsSplitted = []string{}
```
Comment (campoy) [inline PublicGitArchive/pga-create/cmd/pga-create/select.go]: Why is this necessary? Wouldn't the `nil` slice be enough? Then you can do `if len(c.FilterLanguages) > 0` ...

File: PublicGitArchive/pga-create/cmd/pga-create/select.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	StarsFile         string
+	LanguagesFile     string
+	ReposFile         string
+	FilteredLanguages []string
+	MinStars          int
+	TopN              int
+	URLTemplate       string
+}
+
+func selectRepos(parameters selectionParameters) {
```
Comment (campoy) [inline PublicGitArchive/pga-create/cmd/pga-create/select.go]: would it make more sense to have this return an error rather than call `fail` internally?

File: PublicGitArchive/pga-create/cmd/pga-create/select.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+	if err != nil {
+		fail("opening repositories file "+parameters.ReposFile, err)
+	}
+	defer f.Close()
+	gzf, err := gzip.NewReader(f)
+	if err != nil {
+		fail("decompressing repositories file "+parameters.ReposFile, err)
+	}
+	defer gzf.Close()
+	scanner := bufio.NewScanner(gzf)
```
Comment (campoy) [inline PublicGitArchive/pga-create/cmd/pga-create/select.go]: You need to check for `scanner.Err` at the end of this function

File: PublicGitArchive/pga-create/cmd/pga-create/select.go [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+func filterStars(path string, minStars int, topN int, selectedRepos map[int]bool) map[int]bool {
+	f, err := os.Open(path)
+	if err != nil {
+		fail("opening stars file "+path, err)
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	repos := map[int]bool{}
+	var stars int
+	for scanner.Scan() {
```
Comment (campoy) [inline PublicGitArchive/pga-create/cmd/pga-create/select.go]: again, check for `scanner.Err()`

**Comment by smola** [comment]
@campoy Most of this code was just reuse from previous version. I'll add a PR addressing your review.
